src/db/blacklist.rs

```diff
@@ -0,0 +1,84 @@
+use failure::Error;
+use postgres::Connection;
+
+#[derive(Debug, Fail)]
+enum BlacklistError {
+    #[fail(display = "crate {} is already on the blacklist", _0)]
+    CrateAlreadyOnBlacklist(String),
+
+    #[fail(display = "crate {} is not on the blacklist", _0)]
+    CrateNotOnBlacklist(String),
+}
+
+pub fn is_blacklisted(conn: &Connection, name: &str) -> Result<bool, Error> {
+    let rows = conn.query(
+        "SELECT COUNT(*) FROM blacklisted_crates WHERE crate_name = $1;",
```

Comment:
I would use `SELECT COUNT(1) ...` because the SQL Engine does not have to expand `*` to every column in the table. Granted it is only one in this table

Reply:
Can you expand on this? My impression was that `COUNT(*)` doesn't depend on any columns and only counts the rows. There is not much information on this in the postgres docs: https://www.postgresql.org/docs/current/functions-aggregate.html

Reply:
Hi @jyn514 
I readup on todays implementations and I have to retract. 
Todays databases appear to have an optimization for `COUNT(*)` if there are no `NULL` values in the table.
Since you have `NOT NULL` this is fine.
I learned something with this, Thank you!